### PR TITLE
GroupPresentation changes

### PR DESCRIPTION
**Pull request by jwillemsen**
* tests/DCPS/GroupPresentation/subscriber.cpp:

### CODE REVIEW

**Comment by jwillemsen** [comment]
See #623

**Comment by jwillemsen** [comment]
Contains already some improvements, if they are ok, please merge, continue my debugging asap

File: dds/DCPS/DataReaderImpl_T.h [hunk 1]
```diff
@@ -2330,35 +2331,31 @@ class FilterDelayedHandler : public Watchdog {
 
   virtual void reschedule_deadline()
   {
-    if (!data_reader_var_.in()) {
-      return;
-    }
-
-    ACE_GUARD(ACE_Recursive_Thread_Mutex, guard, data_reader_impl_->sample_lock_);
+    if (data_reader_impl_) {
```
Comment (mitza-oci) [inline dds/DCPS/DataReaderImpl_T.h]: Why is this changing from "check a condition and return" to putting the whole method within an "if"?  Seems like the former way was fine.  (Same for cleanup() below.)

File: dds/DCPS/GroupRakeData.cpp [hunk 2]
```diff
@@ -43,11 +43,10 @@ GroupRakeData::get_datareaders(DDS::DataReaderSeq & readers)
   ACE_UNUSED_ARG(readers);
 #ifndef OPENDDS_NO_OBJECT_MODEL_PROFILE
   readers.length(static_cast<CORBA::ULong>(this->sorted_.size()));
-  int i = 0;
+  CORBA::ULong i = 0;
   SortedSet::iterator itEnd = this->sorted_.end();
   for (SortedSet::iterator it = this->sorted_.begin(); it != itEnd; ++it) {
-    readers[i++] =
-      DDS::DataReader::_duplicate(it->si_->instance_state_.data_reader());
+    readers[i++] = DDS::DataReader::_duplicate(it->si_->instance_state_.data_reader());
```
Comment (mitza-oci) [inline dds/DCPS/GroupRakeData.cpp]: This didn't change, no need to reformat the lines.

File: dds/DCPS/DataReaderImpl_T.h (hunk 1 above)
Comment (jwillemsen) [inline dds/DCPS/DataReaderImpl_T.h]: I prefer one exit point

File: dds/DCPS/GroupRakeData.cpp (hunk 2 above)
Comment (jwillemsen) [inline dds/DCPS/GroupRakeData.cpp]: I made a local change which I reverted

**Pull request merged**